tests/Toolkit/ATest.php

```diff
@@ -209,36 +209,140 @@ public function testMapWithClassMethod()
-
-		// append feature
-		$this->assertSame(['a', 'b', 'c', 'd', 'a'], A::merge([1 => 'a', 4 => 'b'], [1 => 'c', 3 => 'd', 5 => 'a']));
-		$this->assertSame(['a', 'b', 'c', 'd', 'a'], A::merge([1 => 'a', 4 => 'b'], [1 => 'c', 3 => 'd', 5 => 'a'], true));
-		$this->assertSame([1 => 'c', 4 => 'b', 3 => 'd', 5 => 'a'], A::merge([1 => 'a', 4 => 'b'], [1 => 'c', 3 => 'd', 5 => 'a'], false));
-		$this->assertSame(['a' => ['b', 'c', 'e', 'd']], A::merge(['a' => [1 => 'b', 4 => 'c']], ['a' => [1 => 'e', 3 => 'd']], true));
-		$this->assertSame(['a' => [1 => 'c', 4 => 'b', 3 => 'd', 5 => 'a']], A::merge(['a' => [1 => 'a', 4 => 'b']], ['a' => [1 => 'c', 3 => 'd', 5 => 'a']], false));
-
-		// replace feature
-		$a = [
-			'a' => ['a', 'b', 'c']
-		];
+		$a        = ['a' => ['b', 'c']];
+		$b        = ['a' => ['b', 'd']];
+		$expected = ['a' => ['b', 'c', 'b', 'd']];
+		$result   = A::merge($a, $b);
+		$this->assertSame($expected, $result);
 
-		$b = [
-			'a' => ['d', 'e', 'f']
-		];
+		$a        = ['a' => ['b' => 'c', 'd' => 'e']];
+		$b        = ['a' => ['b' => 'd']];
+		$expected = ['a' => ['b' => 'd', 'd' => 'e']];
+		$result   = A::merge($a, $b);
+		$this->assertSame($expected, $result);
+
+		$a        = ['a' => 'b'];
+		$b        = ['a' => ['c', 'd']];
+		$expected = ['a' => ['c', 'd']];
+		$result   = A::merge($a, $b);
+		$this->assertSame($expected, $result);
+
+		$a        = ['a' => ['c', 'd']];
+		$b        = ['a' => 'b'];
+		$expected = ['a' => 'b'];
+		$result   = A::merge($a, $b);
+		$this->assertSame($expected, $result);
+	}
+
+	/**
+	 * @covers ::merge
+	 */
+	public function testMergeMultiples()
+	{
+		// simple non-associative arrays
+		$a        = ['a', 'b'];
+		$b        = ['c', 'd'];
+		$c        = ['e', 'f'];
+		$expected = ['a', 'b', 'c', 'd', 'e', 'f'];
+		$result   = A::merge($a, $b, $c);
+		$this->assertSame($expected, $result);
+
+		// simple associative arrays
+		$a        = ['a' => 'b'];
+		$b        = ['c' => 'd'];
+		$c        = ['e' => 'f'];
+		$expected = ['a' => 'b', 'c' => 'd', 'e' => 'f'];
+		$result   = A::merge($a, $b, $c);
+		$this->assertSame($expected, $result);
 
-		$this->assertSame($b, A::merge($a, $b, A::MERGE_REPLACE));
+		// recursive merging
+		$a        = ['a' => ['b', 'c']];
+		$b        = ['a' => ['b', 'd']];
+		$c        = ['a' => ['e'], 'e' => 'f'];
+		$expected = ['a' => ['b', 'c', 'b', 'd', 'e'], 'e' => 'f'];
+		$result   = A::merge($a, $b, $c);
+		$this->assertSame($expected, $result);
+	}
+
+	/**
+	 * @covers ::merge
+	 */
+	public function testMergeModes()
+	{
+		// simple non-associative arrays
+		$a        = [1 => 'a', 4 => 'b'];
+		$b        = [1 => 'c', 3 => 'd', 5 => 'a'];
+
+		// A::MERGE_APPEND
+		$expected = ['a', 'b', 'c', 'd', 'a'];
+		$result   = A::merge($a, $b);
+		$this->assertSame($expected, $result);
+		$result   = A::merge($a, $b);
+		$this->assertSame($expected, $result, true);
+		$result   = A::merge($a, $b);
+		$this->assertSame($expected, $result, A::MERGE_APPEND);
```

Comment:
I think the last argument was supposed to be passed to `A::merge()`, not the assertion, right?

Reply:
I just fixed it.